### ISSUE TITLE
Is it possible to manipulate/transform the Html type

### ISSUE THREAD

**Issue by mrkkrp** [jaspervdj/blaze-html#102]
Hi, I work with libraries that convert stuff into `Html` or `Markup` types. Now I need to do additional processing on that, for example I would like to transform all links in HTML document (produced by something like `Text -> Html`) generated from markdown or add anchors to article sections server-side. I don't see how `Html` can be transformed with the library. How in general I should approach this sort of task?

**Comment by meiersi** [comment]
Hi Mark, blaze-html's `Html` type is write only and not meant to be
analysed. For your analysis I'd suggest a library like tagsoup or
html-conduit. If you also have well-formed xml input, then xml-conduit is
your friend. (Sry..no links..am on the phone)

Note that the conduit libraries might have a bit a higher initial learning
curve, but do provide you also with the ability to perform streaming
transformations.

Mark Karpov <notifications@github.com> schrieb am Sa., 5. Dez. 2015 10:14
AM:

**Comment by mrkkrp** [comment]
@meiersi, Thanks for the answer, `tagsoup` seems to be for extraction of information, not for transformation. Anyway I got it, obviously we should “patch” data before it gets into `Html`. `markdown` package has [an option](http://haddock.stackage.org/nightly-2015-12-05/markdown-0.1.13.2/Text-Markdown.html#v:msBlockFilter) to transform blocks of data before they are written to `Html`, but I have yet to figure out how to use it given that `Block` and `Inline` types are not exported! Closing.
